### ISSUE TITLE
A `.detach()` is missing in `MergeGrowingModule.forward`

### ISSUE THREAD

**Issue by TheoRudkiewicz** [growingnet/gromo#135]
We do not detach `self.activity`. This will create a memory leak.

Original:
```python
    def forward(self, x: torch.Tensor) -> torch.Tensor:
        for t in (self.tensor_s, self.previous_tensor_s, self.previous_tensor_m):
            if t:
                t.updated = False

        if self.store_input > 0:
            self.input = x
            self.input.retain_grad()

        if self.post_merge_function and (x is not None):
            y = self.post_merge_function(x)
        else:
            y = x

        if self.store_activity > 0:
            self.activity = y
            self.tensor_s.updated = False  # reset the update flag

        return y
```

Suggestion:
```python
    def forward(self, x: torch.Tensor) -> torch.Tensor:
        for t in (self.tensor_s, self.previous_tensor_s, self.previous_tensor_m):
            if t:
                t.updated = False

        if self.store_input > 0:
            self.input = x
            self.input.retain_grad()

        if self.post_merge_function and (x is not None):
            y = self.post_merge_function(x)
        else:
            y = x

        if self.store_activity > 0:
            self.activity = y.detach()
            self.tensor_s.updated = False  # reset the update flag

        return y
```

**Comment by TheoRudkiewicz** [comment]
FIxed in #138
